qiskit_experiments/data_processing/nodes.py

```diff
@@ -479,3 +479,45 @@ def _population_error(self, counts_dict) -> Tuple[float, float]:
         p_var = p_mean * (1 - p_mean) / shots
 
         return p_mean, np.sqrt(p_var)
+
+
+class Eigenvalue(DataAction):
+    """Compute eigenvalue from probability."""
```

Comment:
Yes. If you measure qubit in Z-basis, population 0 (1) corresponds to eigenstate of Pauli Z with eigenvalue 1 (-1). This is the same representation with [Sarah 2016](https://arxiv.org/abs/1603.04821) Fig2.

Reply:
Ok. I'd still like a more informative name for this node. `Eigenvalue` is too vague IMO. How about something like `ProbToZEigenvalue` or `PauliEigenvalue`? Also remember that we always physically measure in the Z basis and we only get the other basis by doing a rotation before the measurement in the Z basis.

Reply:
Right, in principle, we can only do Z measurement with our device but once it's projected the outcome should represent the expectation value of the projected basis (up to the error of rotation operations). Since the node to calculate the probability from counts is `Probability` (not `CountToProbability`), `ProbToZEigenvalue` doesn't make sense to me. And I think above calculation doesn't need to be limited to Pauli operators. What do you think of `BasisEigenvalue`?

Reply:
The node name finally becomes `BasisExpectationValue`. See https://github.com/Qiskit/qiskit-experiments/pull/385/commits/ca0e1561c1d9127e77fc229688283d2a98a44f4c